psi4/src/psi4/ccdensity/get_moinfo.cc

```diff
@@ -71,8 +72,8 @@ void get_moinfo(std::shared_ptr<Wavefunction> wfn)
         moinfo.escf = wfn->reference_wavefunction()->reference_energy();
     else
         moinfo.escf = wfn->reference_energy();
-    moinfo.orbspi = wfn->nmopi();
-    moinfo.openpi = wfn->soccpi();
+    moinfo.orbspi = const_cast<psi::Dimension*>(&wfn->nmopi())->pointer();
+    moinfo.openpi = const_cast<psi::Dimension*>(&wfn->soccpi())->pointer();
     moinfo.clsdpi = init_int_array(moinfo.nirreps);
     for(int h = 0; h < moinfo.nirreps; ++h)
         moinfo.clsdpi[h] = wfn->doccpi()[h];
```

Comment:
I wouldn't allow de-consting of pointers, instead use the loop below to copy the data and let the user do what they wish. Just as a general philosophy, I think its best to stop supporting behavior that should be changed so that it is not propagated in the future.